typing/typecore.ml

```diff
@@ -1203,14 +1207,18 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
       raise (Error (loc, !env, Invalid_interval))
   | Ppat_tuple spl ->
       assert (List.length spl >= 2);
-      let spl_ann = List.map (fun p -> (p,newvar ())) spl in
-      let ty = newty (Ttuple(List.map snd spl_ann)) in
+      let spl_ann = List.map (fun p -> (p,newgenvar ())) spl in
+      let ty = newgenty (Ttuple(List.map snd spl_ann)) in
+      begin_def ();
+      let expected_ty = instance expected_ty in
+      end_def ();
+      generalize_structure expected_ty;
```

Comment:
Here it should be OK to use `generalize`: since we are matching a value, different members could be seen as using independent instances.

Reply:
I don't think it makes a difference here: `expected_ty` doesn't contain any type variable at the generic level when we receive it. Which means that after the
```
begin_def;
instance;
end_def;
```
sequence, there won't be any generalizable variables.

Given that, I'd rather keep the call to `generalize_structure` here, for uniformity.

Reply:
Oh, I see. I hadn't seen that due to the `generalize_structure` in `type_cases`, there are no generalized type variables in the type.
This would change if switch later to a more agressive policy of taking full instances, that could be fully generalized soundly.

But then it seems that the only improvement of the current patch is to avoid some warnings with label/constructor disambiguation. Is there any other practical change?